petab/sbml.py

```diff
@@ -0,0 +1,96 @@
+"""Functions for direct access of SBML models"""
+import libsbml
+import math
+
+
+def assignment_rules_to_dict(
+        sbml_model, filter_function=lambda *_: True, remove=False):
+    """
+    Turn assignment rules into dictionary.
+
+    Parameters
+    ----------
+
+    sbml_model:
+        an sbml model instance.
+    filter_function:
+        callback function taking assignment variable as input
+        and returning True/False to indicate if the respective rule should be
+        turned into an observable.
+
+    Returns
+    -------
+
+    A dictionary(assigneeId:{
+        'name': assigneeName,
+        'formula': formulaString
+    })
+    """
+    result = {}
+
+    # iterate over rules
+    for rule in sbml_model.getListOfRules():
+        if rule.getTypeCode() != libsbml.SBML_ASSIGNMENT_RULE:
+            continue
+        assignee = rule.getVariable()
+        parameter = sbml_model.getParameter(assignee)
+        # filter
+        if parameter and filter_function(parameter):
+            result[assignee] = {
+                'name': parameter.getName(),
+                'formula': rule.getFormula()
+            }
+
+    # remove from model?
+    if remove:
+        for parameter_id in result:
+            sbml_model.removeRuleByVariable(parameter_id)
+            sbml_model.removeParameter(parameter_id)
+
+    return result
+
+
+def constant_species_to_parameters(sbml_model):
+    """Convert constant species in the SBML model to constant parameters
```

Comment:
where is this method needed? can one add that in the docstring?

Reply:
Nowhere inside PEtab, but it is quite useful when setting up models for PEtab, since we don't allow for specifying constant species in the condition table.